[quant][graphmode][fx] Add support for qat convbn{relu}1d

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #47332 [quant][graphmode][fx] Test to make sure dequantize node are placed properly
* **#47248 [quant][graphmode][fx] Add support for qat convbn{relu}1d**

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24696524](https://our.internmc.facebook.com/intern/diff/D24696524)

jerryzh168 committed:
[quant][graphmode][fx] Add support for qat convbn{relu}1d

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for qat convbn{relu}1d"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on torch/nn/quantized/modules/conv.py:
```diff
@@ -244,17 +272,7 @@ def from_float(cls, mod):
-        else:
-            activation_post_process = mod.activation_post_process
-        return cls.get_qconv(mod, activation_post_process)
+        return from_float_helper(cls, mod)
```
optional: thoughts on if instead of `from_float_helper` this logic could live in `torch.nn.quantized._ConvNd.from_float`, and then `torch.nn.quantized.Conv3d.from_float` could override it?  Seems like that is what's actually happening, but it's not obvious from the current names

jerryzh168 replied:
oh sure. will move that function to torch.nn.quantized._ConvNd.from_float

vkuzo reviewed (approved):
lg, some optional comments inline.  Should we also add a numerical correctness test?

jerryzh168 commented:
right now most of the ops are not tested. theoretically as long as the observers are inserted correctly and float -> quantized transformation is correct the numerics should match. we have tests for both of the subparts. maybe we can use real models (e.g. torchvision) as e2e tests to test numerics in the future.

jerryzh168 commented:
ok, I think we can change the test to test numerics too

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for qat convbn{relu}1d"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24696524](https://our.internmc.facebook.com/intern/diff/D24696524)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for qat convbn{relu}1d"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24696524](https://our.internmc.facebook.com/intern/diff/D24696524)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for qat convbn{relu}1d"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24696524](https://our.internmc.facebook.com/intern/diff/D24696524)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for qat convbn{relu}1d"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24696524](https://our.internmc.facebook.com/intern/diff/D24696524)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for qat convbn{relu}1d"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24696524](https://our.internmc.facebook.com/intern/diff/D24696524)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for qat convbn{relu}1d"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24696524](https://our.internmc.facebook.com/intern/diff/D24696524)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for qat convbn{relu}1d"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24696524](https://our.internmc.facebook.com/intern/diff/D24696524)

[ghstack-poisoned]
